tests/test_facebook_discovery.py

```diff
@@ -0,0 +1,154 @@
+                        set(stream_properties[0].get(
+                            "metadata", {self.PRIMARY_KEYS: None}).get(self.PRIMARY_KEYS, []))))
+
+                actual_replication_method = stream_properties[0].get(
+                    "metadata", {self.REPLICATION_METHOD: None}).get(self.REPLICATION_METHOD)
+
+
+                # BUG_2 | https://stitchdata.atlassian.net/browse/SRCE-4856
+                failing_with_no_replication_keys = {
+                    'ads_insights_country', 'adsets', 'adcreative', 'ads', 'ads_insights_region',
+                    'campaigns', 'ads_insights_age_and_gender', 'ads_insights_platform_and_device',
+                    'ads_insights_dma', 'ads_insights'
+                }
+                if stream not in failing_with_no_replication_keys:  # BUG_2
+                    # verify the actual replication matches our expected replication method
+                    self.assertEqual(
+                        self.expected_replication_method().get(stream, None),
+                        actual_replication_method,
+                        msg="The actual replication method {} doesn't match the expected {}".format(
+                            actual_replication_method,
+                            self.expected_replication_method().get(stream, None)))
+
+                    # verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
+                    if stream_properties[0].get(
+                            "metadata", {self.REPLICATION_KEYS: []}).get(self.REPLICATION_KEYS, []):
+                        self.assertTrue(actual_replication_method == self.INCREMENTAL,
+                                        msg="Expected INCREMENTAL replication "
+                                            "since there is a replication key")
+                    else:
+                        self.assertTrue(actual_replication_method == self.FULL_TABLE,
+                                        msg="Expected FULL replication since there is no replication key")
+
+
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_automatic_fields = expected_primary_keys | expected_replication_keys
+
+                # verify that primary, replication and foreign keys
+                # are given the inclusion of automatic in annotated schema.
+                actual_automatic_fields = {key for key, value in schema["properties"].items()
+                                           if value.get("inclusion") == "automatic"}
+                self.assertEqual(expected_automatic_fields, actual_automatic_fields)
+
+                self.assertTrue(
+                    all({value.get("inclusion") == "available" for key, value
+                         in schema["properties"].items()
+                         if key not in actual_automatic_fields}),
```

Comment:
I like the set comprehension here. We don't expect `actual_automatic_fields` to ever be that big, but for a big enough set I bet there's a little speed boost